design/priority-class-name-support_design.md

```diff
@@ -0,0 +1,162 @@
+        Template: corev1api.PodTemplateSpec{
+            // ... existing fields ...
+            Spec: corev1api.PodSpec{
+                // ... existing fields ...
+                PriorityClassName: veleroutil.GetPriorityClassNameFromVeleroServer(deployment),
+            },
+        },
+    },
+}
+```
+
+A new function, `GetPriorityClassNameFromVeleroServer`, will be added to the `veleroutil` package to retrieve the priority class name from the Velero server deployment:
+
+```go
+func GetPriorityClassNameFromVeleroServer(deployment *appsv1api.Deployment) string {
+    return deployment.Spec.Template.Spec.PriorityClassName
+}
+```
+
+## Alternatives Considered
+1. **Using separate flags for each component**: We could have added separate flags for the Velero deployment, node agent daemonset, and maintenance jobs. However, this would have increased the complexity of the CLI and would have been more difficult to maintain.
+
+2. **Using a configuration file**: We could have added support for specifying the priority class name in a configuration file. However, this would have required additional changes to the Velero CLI and would have been more complex to implement.
+
+## Security Considerations
+There are no security considerations for this feature.
+
+## Compatibility
+This feature is compatible with all Kubernetes versions that support priority classes. The PodPriority feature became stable in Kubernetes 1.14. For more information, see the [Kubernetes documentation on Pod Priority and Preemption](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/).
+
+## Implementation
+The implementation will involve the following steps:
+1. Add the `PriorityClassName` field to the `VeleroOptions` struct
+2. Add the `priorityClassName` field to the `podTemplateConfig` struct
+3. Add the `WithPriorityClassName` function
+4. Update the `Deployment` function to use the priority class name
+5. Update the `DaemonSet` function to use the priority class name
+6. Add the `GetPriorityClassNameFromVeleroServer` function to the `veleroutil` package
+7. Update the maintenance job creation to use the priority class name from the Velero server deployment
+8. Add the `--priority-class-name` flag to the `velero install` command
+
+Note: We're only adding the flag to the CLI install command, not to the server configuration, as the priority class name is only needed during installation. This is because:
+
+1. The priority class name is set on the Velero deployment and node agent daemonset during installation.
+2. For maintenance jobs, the priority class name is inherited from the Velero server deployment through the `GetPriorityClassNameFromVeleroServer` function in the `veleroutil` package.
+3. For data mover pods created by the exposer, the priority class name is inherited from the node agent daemonset through the `getInheritedPodInfo` function, which calls `nodeagent.GetPodSpec` to get the pod spec from the node agent daemonset.
```

Comment:
Data movers should be with low priority in most cases, in another word, when resource is limited, data movers should not run.

If we decide to inherit from node-agent, we should make sure the priority class configuration is separate for node-agent and Velero server. cc @ywk253100